src/topotoolbox/gridmixins/fillsinks.py

```diff
@@ -0,0 +1,18 @@
+from .._grid import grid_fillsinks
+
+import numpy as np
+import copy
+
+class FillsinksMixin():
+    def fillsinks(self):
+
+        dem = self.z.astype(np.float32)
+
+        output = np.zeros_like(dem)
+
+        grid_fillsinks(output, dem, self.rows, self.columns)
+
+        result = copy.copy(self)
+        result.z = output.copy()
```

Comment:
I might be wrong, but I don't think you need this copy. If you tried to assign `result.z[0,0]  = output[0,0]` and so on, you would overwrite the elements of `self.z`, which we don't want to do. However, I believe `result.z = output` just makes `result.z` point at `output` without touching `self.z`, so it should be safe without the copy.

Reply:
I think you may be right. The use `.copy()` seems to be an artifact from earlier implementations. We should be able to safely remove it.